src/DefaultUnleash.php

```diff
@@ -59,21 +63,157 @@ public function isEnabled(string $featureName, ?Context $context = null, bool $d
+
+        return $this->getVariantForFeature($feature, $context, $dependencies, $fallbackVariant);
+    }
+
+    public function register(): bool
+    {
+        return $this->registrationService->register($this->strategyHandlers);
+    }
+
+    /**
+     * Finds a feature with it's parent features. Posts events if the feature is not found.
+     *
+     * @param string  $featureName name of the feature to find
+     * @param Context $context     the context to use
+     *
+     * @return null|array{
+     *   feature: Feature|null,
+     *   dependencies: array<string, Feature>,
+     * }
+     */
+    public function findFeatureAndDependencies(string $featureName, ?Context $context): ?array
+    {
+        $features = $this->repository->getFeatures();
+        assert(is_array($features));
+        $context ??= $this->configuration->getContextProvider()->getContext();
+
+        $feature = $features[$featureName] ?? null;
+
+        if ($feature === null) {
+            $event = new FeatureToggleNotFoundEvent($context, $featureName);
+            $this->configuration->getEventDispatcherOrNull()?->dispatch(
+                $event,
+                UnleashEvents::FEATURE_TOGGLE_NOT_FOUND,
+            );
+
+            return [
+                'feature' => null,
+                'dependencies' => [],
+            ];
+        }
+
+        $dependencyDefinitions = $feature->getDependencies();
+
+        if ($dependencyDefinitions === null) {
+            return [
+                'feature' => $feature,
+                'dependencies' => [],
+            ];
+        }
+
+        $dependencies = [];
+        foreach ($dependencyDefinitions as $dependencyDefinition) {
+            $name = $dependencyDefinition->getFeature();
+            $dependency = $features[$name] ?? null;
+            if ($dependency !== null) {
+                $dependencies[$name] = $dependency;
+            } else {
+                $event = new FeatureToggleDependencyNotFoundEvent($context, $name);
+                $this->configuration->getEventDispatcherOrNull()?->dispatch(
+                    $event,
+                    UnleashEvents::FEATURE_TOGGLE_NOT_FOUND,
+                );
+            }
+        }
+
+        return [
+            'feature' => $feature,
+            'dependencies' => $dependencies,
+        ];
+    }
+
+    /**
+     * Checks if parent feature flag requirement is satisfied.
+     *
+     * @param Dependency $dependency    the dependency to check
+     * @param Feature    $parentFeature the parent feature to check
+     * @param Context    $context       the context to use
+     */
+    public function isDependencySatisfied(
+        ?Dependency $dependency = null,
+        ?Feature $parentFeature = null,
+        ?Context $context = null,
+    ): bool {
+        if ($dependency === null) {
+            return true;
+        }
+        $context ??= $this->configuration->getContextProvider()->getContext();
```

Comment:
you're passing $context in line 155 and now you're reassigning the same variable here. Is it intentional?

Reply:
It's not a reassignment, it's the `null coalescing assign operator` (catchy name, innit?). It's basically shorthand for this:

```php
if (!isset($context)) {
  $context = $this->configuration->getContextProvider()->getContext();
}
```

Reply:
@RikudouSage you're right, my bad :)